StitchCore/StitchCore/Source/Core/Storage.swift

```diff
@@ -0,0 +1,132 @@
+import Foundation
+
+private let latestVersion = 1
+private let versionKey = "__stitch_storage_version__"
+
+/// Run a migration on the currently used storage
+/// that checks to see if the current version is up to date.
+/// If the version has not been set, this method will migrate
+/// to the latest version.
+/// - parameter suiteName: namespace to key with
+/// - parameter storage: storage to check for version key
+internal func runMigration(suiteName: String, storage: inout Storage) {
+    switch storage.value(forKey: versionKey) {
+    case .none:
+    #if !os(Linux)
+        // map each of the store's keys to a Promise
+        // that fetches the each value for each key,
+        // sets the old value to the new "namespaced" key
+        // remove the old key value pair,
+        // and set the version number
+        // we only care about UserDefaults here since no other
+        // `Storage` type was available before version 1
+        guard var originalStorage = UserDefaults.init(suiteName: "com.mongodb.stitch.sdk.UserDefaults") else {
+            return
+        }
+
+        // legacy keys
+        ["StitchCoreAuthJwtKey",
+         "StitchCoreAuthRefreshTokenKey",
+         "com.mongodb.stitch.sdk.authentication",
+         "StitchCoreIsLoggedInUserDefaultsKey"].forEach { key in
+            storage[key] = originalStorage[key]
+            originalStorage[key] = nil
+        }
+        
+        storage[versionKey] = latestVersion
+    #endif
+    // in future versions, `case 1:`, `case 2:` and so on
+    // could be added to perform similar migrations
+    default: break
+    }
+}
+
+internal struct StorageKeys {
+    internal let isLoggedInUDKey: String
```

Comment:
[q] Given the struct is internal, does that not make the internal props internal too? Also, isn't internal the default access anyway?

Reply:
This ends up being a style choice.